plotter/histogram_gnuplotstyle_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image/color"
+	"log"
+	"math"
+	"math/rand"
+	"testing"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// An example of making a histogram with a custom style.
+func ExampleHistogramGnuplot() {
+	// stdNorm returns the probability of drawing a
+	// value from a standard normal distribution.
+	stdNorm := func(x float64) float64 {
+		const sigma = 1.0
+		const mu = 0.0
+		const root2π = 2.50662827459517818309
+		return 1.0 / (sigma * root2π) * math.Exp(-((x-mu)*(x-mu))/(2*sigma*sigma))
+	}
+
+	n := 10000
+	vals := make(Values, n)
+	for i := 0; i < n; i++ {
+		vals[i] = rand.NormFloat64()
+	}
+
+	p, err := plot.New()
+	if err != nil {
+		log.Panic(err)
+	}
+	p.Title.Text = "Histogram"
+	p.X.Label.Text = "X-axis"
+	p.Y.Label.Text = "Y-axis"
+	p.Style = GnuplotStyle{}
+
+	h, err := NewHist(vals, 16)
+	if err != nil {
+		log.Panic(err)
+	}
+	h.Normalize(1)
+	p.Add(h)
+
+	// The normal distribution function
+	norm := NewFunction(stdNorm)
+	norm.Color = color.RGBA{R: 255, A: 255}
+	norm.Width = vg.Points(2)
+	p.Add(norm)
+
+	err = p.Save(200, 200, "testdata/histogram-gnuplot-style.png")
+	if err != nil {
+		log.Panic(err)
+	}
+}
+
+func TestHistogramGnuplot(t *testing.T) {
+	checkPlot(ExampleHistogramGnuplot, t, "histogram-gnuplot-style.png")
+}
+
+type GnuplotStyle struct{}
+
+func (s GnuplotStyle) DrawPlot(p *plot.Plot, c draw.Canvas) {
+	if p.BackgroundColor != nil {
+		c.SetColor(p.BackgroundColor)
+		c.Fill(c.Rectangle.Path())
+	}
+
+	{
+		xpad := p.X.Padding
+		ypad := p.Y.Padding
+		defer func() {
+			p.X.Padding = xpad
+			p.Y.Padding = ypad
+		}()
+	}
+
+	p.X.Padding = 0
```

Comment:
FWIW, for my quick pie-in-the-sky proposal, I put them in the `vg/draw` package because all the other `XyzStyle` (`draw.TextStyle`, `draw.LineStyle`,...) were there :)